transaction/core/src/account_keys.rs

```diff
@@ -110,26 +143,38 @@ impl PublicAddress {
 
     /// Get the optional fog url.
     pub fn fog_url(&self) -> Option<&str> {
-        self.fog_url.as_deref()
+        if self.fog_url.is_empty() {
+            None
+        } else {
+            Some(&self.fog_url)
+        }
     }
 }
 
 /// Complete AccountKey, containing the pair of secret keys, which can be used
 /// for spending, and optionally some fog-related info,
 /// can be used for spending. This should only ever be present in client code.
-#[derive(Clone, Serialize, Deserialize, Message)]
+#[derive(Clone, Message)]
 pub struct AccountKey {
     /// Private key 'a' used for view-key matching.
     #[prost(message, required, tag = "1")]
-    view_private_key: RistrettoPrivate,
+    pub view_private_key: RistrettoPrivate,
 
     /// Private key `b` used for spending.
     #[prost(message, required, tag = "2")]
-    spend_private_key: RistrettoPrivate,
+    pub spend_private_key: RistrettoPrivate,
 
     /// Fog URL (if user has Fog service)
```

Comment:
Could you add to the comment to mention what the value will be if the user doesn't have a Fog service?